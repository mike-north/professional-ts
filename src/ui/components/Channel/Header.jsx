import * as React from 'react';

const Header = ({ title, description }) => (
  <header className="border-b flex px-6 py-2 items-center flex-row channel-header">
    <div className="flex-1">
      <h3 className="text-gray-800 mb-1 font-extrabold channel-header__title">
        <span aria-hidden="true">#</span>
        {title}
      </h3>
      <h4 className="text-gray-600 text-sm truncate channel-header__description">
        {description}
      </h4>
    </div>
  </header>
);

export default Header;
