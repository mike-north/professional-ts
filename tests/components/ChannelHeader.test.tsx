import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Header from '../../src/ui/components/Channel/Header';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Header title="Recruiting channel" description="Find a job here!" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
