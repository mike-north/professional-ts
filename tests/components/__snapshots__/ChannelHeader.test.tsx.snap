// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<header
  className="border-b flex px-6 py-2 items-center flex-row channel-header"
>
  <div
    className="flex-1"
  >
    <h3
      className="text-gray-800 mb-1 font-extrabold channel-header__title"
    >
      <span
        aria-hidden="true"
      >
        #
      </span>
      Recruiting channel
    </h3>
    <h4
      className="text-gray-600 text-sm truncate channel-header__description"
    >
      Find a job here!
    </h4>
  </div>
</header>
`;
